src/yaml.cr

```diff
@@ -147,4 +148,28 @@ module YAML
   def self.dump(object, io : IO)
     object.to_yaml(io)
   end
+
+  # Determines if the scaler is reserved
+  def self.reserved_scalar(value : String)
```

Comment:
@ysbaddaden I am not sure I understand. `infinity` and `NaN are represented with `.nan` and `.inf` how do you mean int/float will have to be parsed again?